src/drt/src/ta/FlexTA_init.cpp

```diff
@@ -310,6 +317,103 @@ void FlexTAWorker::initIroute_helper_generic_helper(frGuide* guide,
   }
 }
 
+void FlexTAWorker::initIroute_helper_generic_fix_endpoints(frGuide* guide,
+                                                           frCoord& minBegin,
+                                                           frCoord& maxEnd)
+{
+  auto [bp, ep] = guide->getPoints();
+  auto net = guide->getNet();
+  bool isH = (getDir() == dbTechLayerDir::HORIZONTAL);
+
+  auto rq = getRegionQuery();
+  vector<frBlockObject*> result;
+
+  Rect box;
+  box = Rect(bp, bp);
+  rq->queryGRPin(box, result);
+  if (!(ep == bp)) {
+    box = Rect(ep, ep);
+    rq->queryGRPin(box, result);
+  }
+
+  frCoord newMinBegin = std::numeric_limits<int>::max();
+  frCoord newMaxEnd = std::numeric_limits<int>::min();
+  for (auto& term : result) {
+    switch (term->typeId()) {
+      case frcInstTerm: {
+        auto iterm = static_cast<frInstTerm*>(term);
+        if (iterm->getNet() != net
+            || !iterm->getInst()->getMaster()->getMasterType().isBlock()) {
```

Comment:
why?

Reply:
I've updated my code and removed this function. It was generating connectivity errors because the iroutes were not connected in some designs.

Also, I've noticed that just adding a check to avoid costing connections to the pins was enough to fix the track assignment and keep the straight connection when possible.